syncall/tw_gtasks_utils.py

```diff
@@ -0,0 +1,174 @@
+        "status",
+        "uuid",
+    ]:  # NOTE(kisseliov): is notes the only place for tw status and description?
+        gtasks_item["notes"] += f"\n* {k}: {tw_item[k]}"
+
+    date_keys = ["scheduled", "due"] if prefer_scheduled_date else ["due", "scheduled"]
+
+    # handle dates  ---------------------------------------------------------------------------
+    # walk through the date_keys using the first of them that's present in the item at hand.
+    # - if 'scheduled' date is prefered we set task due date to scheduled
+    # - if 'due' date is present we set task due date to it
+    # - otherwise we don't set due date at all
+    for date_key in date_keys:
+        if date_key in tw_item.keys():
+            logger.trace(
+                f'Using "{date_key}" date for {tw_item["uuid"]} for setting the due date of'
+                " the task"
+            )
+            gtasks_item["due"] = GTasksSide.format_datetime(
+                GTasksSide.parse_datetime(tw_item[date_key])
+            )
+            break
+
+    # update time
+    if "modified" in tw_item.keys():
+        gtasks_item["updated"] = GTasksSide.format_datetime(
+            GTasksSide.parse_datetime(tw_item["modified"])
+        )
+
+    return gtasks_item
+
+
+def convert_gtask_to_tw(
+    gtasks_item: Item,
+    set_scheduled_date: bool = False,
+) -> Item:
+    """GTasks -> TW Converter.
+
+    If set_scheduled_date, then it will set the "scheduled" date of the produced TW task
+    instead of the "due" date
+    """
+
+    # Parse the description
+    annotations, status, uuid = _parse_gtask_notes(gtasks_item)
+    assert isinstance(annotations, list)
+    assert isinstance(status, str)
+    assert isinstance(uuid, UUID) or uuid is None
+
+    tw_item: Item = {}
+    # annotations
+    tw_item["annotations"] = annotations
+
+    # alias - make aliases dict?
+    if status == "completed":
+        status = "completed"
+
+    if status == "needsAction":
+        status = "pending"
+
+    # Status
+    if status not in ["pending", "completed", "deleted", "waiting", "recurring"]:
+        logger.error(
+            "Invalid status {status} in GCal->TW conversion of item. Skipping status:"
+        )
+    else:
+        tw_item["status"] = status
+
+    # uuid - may just be created -, thus not there
+    if uuid is not None:
+        tw_item["uuid"] = uuid
+
+    # Description
+    tw_item["description"] = gtasks_item["title"]
+
+    # don't meddle with the 'entry' field
+    if set_scheduled_date:
+        date_key = "scheduled"
+    else:
+        date_key = "due"
+
+    end_time = GTasksSide.get_task_completed_time(gtasks_item)
+    tw_item[date_key] = end_time
+
+    # update time
+    if "updated" in gtasks_item.keys():
+        tw_item["modified"] = GTasksSide.parse_datetime(gtasks_item["updated"])
+
+    # NOTE(kisseliov): consider adding custom UDAs for Google Tasks to handle positions and hierarchy
+    return tw_item
+
+
+def _parse_gtask_notes(
```

Comment:
Is this similar to the one we currently have for parsing google calendar tasks? Perhaps abstract this if it is?

Reply:
Created a new function and called it from both the gtasks and gcal paths.